metrics/circonus/circonus.go

```diff
@@ -63,14 +65,30 @@ func (c *Counter) Add(delta float64) { c.m.Add(c.name, uint64(delta)) }
 type Gauge struct {
 	name string
 	m    *circonusgometrics.CirconusMetrics
+	val  float64
+	mtx  sync.RWMutex
 }
 
 // With implements Gauge, but is a no-op, because Circonus metrics have no
 // concept of per-observation label values.
 func (g *Gauge) With(labelValues ...string) metrics.Gauge { return g }
 
 // Set implements Gauge.
-func (g *Gauge) Set(value float64) { g.m.SetGauge(g.name, value) }
+func (g *Gauge) Set(value float64) {
+	g.mtx.Lock()
+	defer g.mtx.Unlock()
+	g.val = value
+	g.m.SetGauge(g.name, value)
+}
+
+// Add implements metrics.Gauge.
+func (g *Gauge) Add(delta float64) {
```

Comment:
I think that this could be better, but couldn't see any way in circonus-gometrics to get the value of a Gauge.